packages/ra-supabase-core/src/authProvider.ts

```diff
@@ -77,6 +77,31 @@ export const supabaseAuthProvider = (
 
             return Promise.resolve();
         },
+        async handleCallback() {
+            const urlSearchParams = new URLSearchParams(
+                window.location.hash.substring(1)
+            );
+
+            const access_token = urlSearchParams.get('access_token');
+            const refresh_token = urlSearchParams.get('refresh_token');
+            const type = urlSearchParams.get('type');
+
+            // Users have reset their password or have just been invited and must set a new password
+            if (type === 'recovery' || type === 'invite') {
+                if (access_token && refresh_token) {
+                    // eslint-disable-next-line no-throw-literal
+                    return {
+                        redirectTo: `set-password?access_token=${access_token}&refresh_token=${refresh_token}&type=${type}`,
+                    };
+                }
+
+                if (process.env.NODE_ENV === 'development') {
+                    console.error(
+                        'Missing access_token or refresh_token for an invite or recovery'
+                    );
+                }
+            }
```

Comment:
This whole section should be factorized between `handleCallback` and `checkAuth` IMO, to ease further maintenance of the code

Reply:
They don't do the exact same thing. One return, the other throws

Reply:
Done